types/yargs/index.d.ts

```diff
@@ -311,6 +313,7 @@ declare namespace yargs {
     type CommandBuilder = { [key: string]: Options } | ((args: Argv) => Argv);
     type SyncCompletionFunction = (current: string, argv: any) => string[];
     type AsyncCompletionFunction = (current: string, argv: any, done: (completion: string[]) => void) => void;
+    type MiddlewareFunction = (args: Arguments) => void;
```

Comment:
That's actually just an inconsistency between the Yargs and DefinitelyTyped.  The two use `args`, `argv`, and `yargs` differently, which causes confusion.   

In this case, the parameter that's passed to the middleware function is [named `innerArgv`](https://github.com/yargs/yargs/blob/6c6c27d8c87c25e052df23abcf5875253b6b20cd/lib/command.js#L234), but this object is _not_ an `Argv` object as defined by DefinitelyTyped.  It's actually [set to `parsed.argv`](https://github.com/yargs/yargs/blob/6c6c27d8c87c25e052df23abcf5875253b6b20cd/lib/command.js#L176)  which [is an `Arguments` object in DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3925a1622a5218568ca23767193b8c69899d0ed1/types/yargs/index.d.ts#L33).  It may also be set to [the result of the `_parseArgs()` method](https://github.com/yargs/yargs/blob/6c6c27d8c87c25e052df23abcf5875253b6b20cd/lib/command.js#L218), which is also [an `Arguments` object in DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3925a1622a5218568ca23767193b8c69899d0ed1/types/yargs/index.d.ts#L145-L146)

Reply:
Indeed. Well explained.